src/sonic_ax_impl/mibs/vendor/cisco/ciscoSwitchQosMIB.py

```diff
@@ -0,0 +1,191 @@
+from enum import unique, Enum
+from bisect import bisect_right
+
+from sonic_ax_impl import mibs
+from ax_interface import MIBMeta, ValueType, MIBUpdater, MIBEntry, SubtreeMIBEntry
+from ax_interface.encodings import ObjectIdentifier
+
+# Maps queue stat counters names to SNMP sub-identifiers.
+CounterNames = {
+    1 : 'ucastSentPkts',
+    2 : 'ucastSentBytes',
+    3 : 'mcastSentPkts',
+    4 : 'mcastSentBytes',
+    5 : 'ucastDroppedPkts',
+    6 : 'ucastDroppedBytes',
+    7 : 'mcastDroppedPkts',
+    8 : 'mcastDroppedBytes' }
+
+# Maps SNMP queue stat counters to SAI counters
+SaiCounterNameMap = {
+    'ucastSentPkts'     : b'SAI_QUEUE_STAT_PACKETS',
+    'ucastSentBytes'    : b'SAI_QUEUE_STAT_BYTES',
+    'mcastSentPkts '    : b'SAI_QUEUE_STAT_PACKETS',
+    'mcastSentBytes'    : b'SAI_QUEUE_STAT_BYTES',
+    'ucastDroppedPkts'  : b'SAI_QUEUE_STAT_DROPPED_PACKETS',
+    'ucastDroppedBytes' : b'SAI_QUEUE_STAT_DROPPED_BYTES',
+    'mcastDroppedPkts'  : b'SAI_QUEUE_STAT_DROPPED_PACKETS',
+    'mcastDroppedBytes' : b'SAI_QUEUE_STAT_DROPPED_BYTES' }
+
+# Maps queue stat counters names to required SAI type of queue
+SaiCounterTypeMap = {
+     'ucastSentPkts'     : b'SAI_QUEUE_TYPE_UNICAST',
+     'ucastSentBytes'    : b'SAI_QUEUE_TYPE_UNICAST',
+     'mcastSentPkts '    : b'SAI_QUEUE_TYPE_MULTICAST',
+     'mcastSentBytes'    : b'SAI_QUEUE_TYPE_MULTICAST',
+     'ucastDroppedPkts'  : b'SAI_QUEUE_TYPE_UNICAST',
+     'ucastDroppedBytes' : b'SAI_QUEUE_TYPE_UNICAST',
+     'mcastDroppedPkts'  : b'SAI_QUEUE_TYPE_MULTICAST',
+     'mcastDroppedBytes' : b'SAI_QUEUE_TYPE_MULTICAST' }
+
+class DirectionTypes(int, Enum):
+    """
+    Queue direction types
+    """
+    INGRESS = 1
+    EGRESS = 2
+
+class QueueStatUpdater(MIBUpdater):
+    """
+    Class to update the info from Counter DB and to handle the SNMP request
+    """
+    def __init__(self):
+        """
+        init the updater
+        """
+        super().__init__()
+        self.db_conn = mibs.init_db()
+        self.reinit_data()
+
+        self.lag_name_if_name_map = {}
+        self.if_name_lag_name_map = {}
+        self.oid_lag_name_map = {}
+
+    def reinit_data(self):
+        """
+        Subclass update interface information
+        """
+        self.port_queues_map, self.queue_stat_map = mibs.init_sync_d_queue_tables(self.db_conn)
+
+    def update_data(self):
+        """
+        Update redis (caches config)
+        Pulls the table references for each interface.
+        """
+        self.reinit_data()
```

Comment:
>reinit_data [](start = 13, length = 11)

the same

Reply:
The queue could be added/removed. Need to update queue map.